packages/mui-lab/src/MonthPicker/MonthPicker.tsx

```diff
@@ -79,8 +81,9 @@ const MonthPicker = React.forwardRef(function MonthPicker<TDate>(
   inProps: MonthPickerProps<TDate>,
   ref: React.Ref<HTMLDivElement>,
 ) {
+  const { inputFormat, renderInput, ...rest } = inProps;
```

Comment:
Since the props aren't supported, they shouldn't be passed.

So the callsite (i.e. usage of `<MonthPicker />`) needs to be updated, not the `MonthPicker`